### ISSUE TITLE
Feature request: create functions for functions that are often called

### ISSUE THREAD

**Issue by danieltroger** [terser/terser#963]
**Feature request**

My 40KB script contains `Object.defineProperty` 29 times, `Object.getOwnPropertySymbols` 22 times and `Object.getOwnPropertyDescriptors` 20 times.

Would be great if terser could create a helper function at the beginning for long functions like this:
`const d = Object.defineProperty.bind(Object)` and then only call d().

Is that doable? It can be hidden behind some `unsafe` option if it has side effects in some edge case :)

**Version (complete output of `terser -V` or specific git commit)**

```
terser 5.5.1

```

**Complete CLI command or `minify()` options used**

Used with parcel v2, .terserrc:
```
{
  "mangle": {
    "eval": true,
  },
  "compress": {
    "unsafe": true,
    "arguments": true,
    "ecma": "2020",
    "passes": 10,
    "unsafe_arrows": true
  },
  "format": {
    "ecma": 2020
  }
}
```

**`terser` input**

Whatever code you like which calls long functions in the global scope often

**`terser` output or error**

Much code

**Expected result**

Less code

**Comment by sirian** [comment]
https://github.com/terser/terser/issues/590

https://github.com/terser/terser/issues/935

**Comment by danieltroger** [comment]
@sirian Thanks for the links, interesting issues! This is not the same though as I suggest assigning a bound function to a variable and call that. I think it would not suffer from the performance concerns mentioned in the other issues?

**Comment by sirian** [comment]
@danieltroger I guess all modern js engines (especially v8) are smart enough too cache/inline constant variables  when on optimization stage (but I'd var changes - there will be deoptimization). You could try run tests like this https://www.measurethat.net/Benchmarks/Show/2207/2/dot-vs-bracket-notation and check. 
If you need more details about optimization/deoptimization - you could check this
https://developers.google.com/web/updates/2012/03/Optimizing-JavaScript
https://v8.dev/docs/turbofan
https://mathiasbynens.be/notes/shapes-ics

**Comment by jridgewell** [comment]
It's not safe to perform this by default, since the value of `Object.defineProperty` could change at any point. Eg, `getOwnPropertySymbols` could need to be polyfilled at some point, and if we had already cached the value of `Object. getOwnPropertySymbols` into a variable then we wouldn't pick up the polyfill.

**Comment by danieltroger** [comment]
@jridgewell fair point, but what if I know it's not going to get polyfilled? Why not put it behind an `unsafe` option?

**Comment by jridgewell** [comment]
We could do an unsafe option, but I'd place it as a lower priority item.
